src/Storages/StorageDistributed.cpp

```diff
@@ -1286,25 +1299,28 @@ void StorageDistributed::initializeFromDisk()
 
     /// Make initialization for large number of disks parallel.
     ThreadPool pool(CurrentMetrics::StorageDistributedThreads, CurrentMetrics::StorageDistributedThreadsActive, CurrentMetrics::StorageDistributedThreadsScheduled, disks.size());
+    std::vector<std::future<void>> futures;
 
     for (const DiskPtr & disk : disks)
     {
-        pool.scheduleOrThrowOnError([&]()
+        auto future = scheduleFromThreadPool<void>([this, disk_to_init = disk]
```

Comment:
where do you add this `future` to `futures`? :thinking: 
same for other 2 places

Reply:
:facepalm: 
Fixed.